packages/devtools_app/lib/src/memory/memory_tracker_model.dart

```diff
@@ -0,0 +1,259 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../table.dart';
+import '../table_data.dart';
+import '../trees.dart';
+import '../utils.dart';
+import 'memory_controller.dart';
+
+class Tracker extends TreeNode<Tracker> {
+  factory Tracker({String className}) {
+    return Tracker._internal(name: className);
+  }
+
+  Tracker._internal({this.name});
+
+  /// Used for both class name and call stack entry value.
+  final String name;
+}
+
+class TrackerClass extends Tracker {
+  TrackerClass(String className) : super._internal(name: className);
+
+  @override
+  String toString() => name;
+}
+
+class TrackerAllocation extends Tracker {
+  TrackerAllocation(this.timestamp) : super._internal(name: null);
+
+  final int timestamp;
+
+  bool displayAll = false;
+
+  /// If displayAll is false then depth of callstack is:
+  final callDepthDisplay = 4;
+
+  @override
+  String toString() => '$timestamp';
+}
+
+class TrackerCall extends Tracker {
+  TrackerCall(String entry) : super._internal(name: entry);
+
+  @override
+  String toString() => name;
+}
+
+class TrackerMore extends Tracker {
+  TrackerMore(this.stacktrace) : super._internal(name: 'More ...');
+
+  final List<String> stacktrace;
+
+  @override
+  String toString() => name;
+}
+
+class _TrackerClassColumn extends TreeColumnData<Tracker> {
+  _TrackerClassColumn() : super('Tracking');
+
+  @override
+  dynamic getValue(Tracker dataObject) {
+    if (dataObject is TrackerClass || dataObject is TrackerMore) {
+      return dataObject.name;
+    } else if (dataObject is TrackerCall) {
+      final TrackerCall call = dataObject;
+      return call.name;
+    } else if (dataObject is TrackerAllocation) {
+      final TrackerAllocation allocation = dataObject;
+      return '${dataObject.parent.name} @ ${prettyTimestamp(allocation.timestamp)}';
+    }
+
+    assert(false, 'Unknown dataObject');
+  }
+
+  @override
+  int compare(Tracker a, Tracker b) {
+    Comparable valueA = getValue(a);
+    Comparable valueB = getValue(b);
+    if (a is TrackerCall) {
+      // Sort is always by child index for the stacktrace, call stack should not reorder.
+      valueA = a.children.indexOf(a);
```

Comment:
this seems odd. why does the children of a contain a? Add a comment explaining more what is going on.
Naively seems like we are just returning -1 a lot.

Reply:
Yes it is odd in many ways.  Returning -1 happens to not sort (which is what I wanted).  Also,  `valueB = b.children.indexOf(a)` is wrong too.  Must have been a brain fart.

The two lines should have been:

```
      valueA = a.parent.children.indexOf(a);
      valueB = b.parent.children.indexOf(b);
```